src/Adapters/controllers/AddonController.java

```diff
@@ -0,0 +1,80 @@
+package Adapters.controllers;
+
+import UseCases.Orders.CreateOrderInputBoundary;
+import businessrules.addon.inputboundaries.CreateAddonInputBoundary;
+import businessrules.addon.inputboundaries.DeleteAddonInputBoundary;
+import businessrules.addon.inputboundaries.ReadAddonInputBoundary;
+import businessrules.addon.inputboundaries.UpdateAddonInputBoundary;
+import org.json.JSONObject;
+
+public class AddonController {
+    CreateAddonInputBoundary createAddonInputBoundary;
+    DeleteAddonInputBoundary deleteAddonInputBoundary;
+    ReadAddonInputBoundary readAddonInputBoundary;
+    UpdateAddonInputBoundary updateAddonInputBoundary;
+
+    public AddonController(CreateAddonInputBoundary createAddonInputBoundary,DeleteAddonInputBoundary deleteAddonInputBoundary,
+                           ReadAddonInputBoundary readAddonInputBoundary,  UpdateAddonInputBoundary updateAddonInputBoundary){
+
+        this.createAddonInputBoundary = createAddonInputBoundary;
+        this.deleteAddonInputBoundary = deleteAddonInputBoundary;
+        this.readAddonInputBoundary = readAddonInputBoundary;
+        this.updateAddonInputBoundary = updateAddonInputBoundary;
+
+    }
+
+    public void runUpdateAddon(String input){
+        JSONObject update_data = new JSONObject(input);
+        if(!(update_data.has("vendorToken") && update_data.has("addonID") && update_data.has("addonObject"))){
```

Comment:
Im not sure if this statement will check what you think. I think it will check if update does not have vendortoken and updatedata has the other 2 attributes.

Reply:
There is a parentheses around each all and statements. Some of the other checks didn't have this but I have just added them.